src/tests/integration/test_generic_db_operations.py

```diff
@@ -1095,3 +1095,49 @@ def test_autocommit_create_table_multiple_cells(
     ).result
 
     assert len(result) == 3
+
+
+@pytest.mark.parametrize("ip_with_dynamic_db", ALL_DATABASES)
+def test_explore_fetch_sql_with_pagination(
+    ip_with_dynamic_db,
+    request,
+    tmp_empty,
+):
+    if ip_with_dynamic_db not in [
+        "ip_with_postgreSQL",
+        "ip_with_duckDB",
+        "ip_with_SQLite",
+    ]:
+        pytest.xfail(reason="There may be an issue due to code refactoring")
+
+    ip_with_dynamic_db = request.getfixturevalue(ip_with_dynamic_db)
+
+    ip_with_dynamic_db.run_cell(
+        """
+    %%sql
+    CREATE TABLE people (name varchar(50), number int, country varchar(50));
+    INSERT INTO people VALUES ('joe', 82, 'usa');
+    INSERT INTO people VALUES ('paula', 93, 'uk');
+    INSERT INTO people VALUES ('sam', 77, 'canada');
+    INSERT INTO people VALUES ('emily', 65, 'usa');
+    INSERT INTO people VALUES ('michael', 89, 'usa');
+    INSERT INTO people VALUES ('sarah', 81, 'uk');
+    INSERT INTO people VALUES ('james', 76, 'usa');
+    INSERT INTO people VALUES ('angela', 88, 'usa');
+    INSERT INTO people VALUES ('robert', 82, 'usa');
+    INSERT INTO people VALUES ('lisa', 92, 'uk');
+    INSERT INTO people VALUES ('mark', 77, 'usa');
+    INSERT INTO people VALUES ('jennifer', 68, 'australia');
+    """
+    )
+    ip_with_dynamic_db.run_cell(
+        """
+        %%sql --save citizen
+        select * from people where country = 'usa'
+        """
+    )
```

Comment:
we're already creating some tables for all databases (although they have different names), check out how we're getting the table names. there's no need to create new tables here